modelhub/tests_modelhub/functional/modelhub/test_stack_extracted_contexts.py

```diff
@@ -0,0 +1,218 @@
+    if not is_bigquery(engine):
+        return pd.DataFrame(parsed_data)
+
+    bq_data = []
+    for event in parsed_data:
+        taxonomy_data = [
+            {
+                '_type': event['value']['_type'],
+                '_types': json.dumps(event['value']['_types']),
+                'global_contexts': json.dumps(event['value']['global_contexts']),
+                'location_stack': json.dumps(event['value']['location_stack']),
+                'time': event['value']['time'],
+                'event_id': str(event['event_id']),
+                'cookie_id': str(event['cookie_id']),
+            }
+        ]
+        bq_data.append({'contexts_io_objectiv_taxonomy_1_0_0': taxonomy_data})
+
+    return pd.DataFrame(bq_data)
+
+
+def get_expected_context_pandas_df(engine) -> pd.DataFrame:
+    pdf = pd.DataFrame(get_parsed_objectiv_data(engine))
+    context_pdf = pdf['value'].apply(pd.Series)
+
+    context_pdf['event_id'] = pdf['event_id']
+    context_pdf['day'] = pdf['day']
+    context_pdf['moment'] = pdf['moment']
+    context_pdf['user_id'] = pdf['cookie_id']
+    context_pdf = context_pdf.rename(
+        columns={
+            '_type': 'event_type',
+            '_types': 'stack_event_types',
+        }
+    )
+
+    return context_pdf[_EXPECTED_CONTEXT_COLUMNS]
+
+
+def _get_extracted_contexts_pipeline(db_params) -> ExtractedContextsPipeline:
+    engine = create_engine_from_db_params(db_params)
+    return ExtractedContextsPipeline(engine=engine, table_name=db_params.table_name)
+
+
+def test_get_pipeline_result(db_params) -> None:
+    context_pipeline = _get_extracted_contexts_pipeline(db_params)
+    engine = context_pipeline._engine
+
+    result = context_pipeline().sort_values(by='event_id').to_pandas()
+    expected = get_expected_context_pandas_df(engine)
+    pd.testing.assert_frame_equal(expected, result)
+
+
+def test_get_initial_data(db_params) -> None:
+    context_pipeline = _get_extracted_contexts_pipeline(db_params)
+    engine = context_pipeline._engine
+    expected = _get_parsed_test_data_pandas_df(engine)
+    result = context_pipeline._get_initial_data()
+
+    if not is_bigquery(engine):
+        assert set(result.data_columns) == set(expected.columns)
+        sorted_columns = sorted(result.data_columns)
+        result = result[sorted_columns].sort_values(by='event_id')
+        assert_equals_data(
+            result,
+            expected_columns=sorted_columns,
+            expected_data=expected[sorted_columns].to_numpy().tolist(),
+            use_to_pandas=True,
+        )
+        return
+
+    taxonomy_column = 'contexts_io_objectiv_taxonomy_1_0_0'
+    assert taxonomy_column in result.data
+    assert isinstance(result[taxonomy_column], bach.SeriesList)
+
+    # need to sort the rows since order is non-deterministic
+    result['event_id'] = result[taxonomy_column].elements[0].elements['event_id']
+    result = result.sort_values(by='event_id')
+    result = result.materialize()
```

Comment:
```suggestion
    result = result.materialize()
    result = result.sort_values(by='event_id')
```
`materialize()` undoes the effect of `sort_values()`